plugins/catalog-backend/src/next/database/DefaultProcessingDatabase.ts

```diff
@@ -273,14 +274,27 @@ export class DefaultProcessingDatabase implements ProcessingDatabase {
       );
     }
 
-    if (toAdd.length) {
-      for (const { entity, locationKey } of toAdd) {
+    if (toUpsert.length) {
+      for (const {
+        deferred: { entity, locationKey },
+        hash,
+      } of toUpsert) {
         const entityRef = stringifyEntityRef(entity);
 
         try {
-          let ok = await this.insertUnprocessedEntity(tx, entity, locationKey);
+          let ok = await this.updateUnprocessedEntity(
```

Comment:
So the order of these can of course be debated. Or even, whether they might be two entirely different methods. My assumption was that it would be more common that already-emitted things had changes, than being either added or having a change of location key.